File: Boids_2_c_raylib_myedit/src/game.c
#include <stdlib.h>
#include <math.h>
#include <raylib.h>
#include <raymath.h>
#include "aabb.h"
#include "boid.h"
#include "game.h"

GameModel model;

void Game_Init()
{
    SetRandomSeed(0);
    model.aabb.p0 = (Vector2){ 0, 0 };
    model.aabb.p1 = (Vector2){ 50, 50 };
	for (int i = 0; i < GAME_BOIDSCOUNT; i++)
		model.boids[i] = CreateBoidRand(model.aabb);
}
void Game_Update(double deltaTime)
{
    for (int i = 0; i < GAME_BOIDSCOUNT; i++)
        UpdateBoid(model.boids[i], model.boids, GAME_BOIDSCOUNT, model.aabb, deltaTime);
}
void Game_End()
{
    for (int i = 0; i < GAME_BOIDSCOUNT; i++)
		FreeBoid(model.boids[i]);
}
GameModel* Game_GetModel() { return &model; }

File: Boids_C#/src/Game.cs
using MyVector2;

public class Game
{
    public AABB aabb;
    public Boid[] boids = new Boid[300];

    public Game()
    {
        var subgen = new Subgen(0);
        aabb = new AABB(Vector2.Zero, new Vector2(50,50));
        for (int i = 0; i < boids.Length; i++)
            boids[i] = new Boid(aabb,subgen);
    }
    public static void Update(Game gameState, long deltaTime)
    {
        var deltaTimeInSeconds = ((float)deltaTime)/1000f;
        Boid.Update(gameState.boids, ref gameState.aabb, deltaTimeInSeconds);
    }
}